File: OOB class/proj1_cpp/wilcze_jagody.cpp
#include "wilcze_jagody.h"
#include "Swiat.h"

wilcze_jagody::wilcze_jagody(short x, short y, Swiat* swiat, short wiek) {
	this->wiek = wiek;
	this->nazwa = "Wilcze Jagody";
	this->inicjatywa = 0;
	this->sila = 99;
	this->znak = 'W';
	this->polozenie[OX] = x;
	this->polozenie[OY] = y;
	this->swiat = swiat;
	swiat->setJesliWolnePole(x, y, znak);
}


void wilcze_jagody::zasiej(short x, short y) {
	swiat->addOrganizm(new wilcze_jagody(x, y, this->swiat, swiat->get_and_incrementKtoryUrodzony()));
}

int wilcze_jagody::kolizja(Organizm* napotkany_org) {
	return POISON;
}

File: OOB class/proj2_java/src/symulator/rosliny/wilcze_jagody.java
package symulator.rosliny;

import symulator.Organizm;
import symulator.Roslina;
import symulator.Swiat;

public class wilcze_jagody extends Roslina {
    public wilcze_jagody(Swiat swiat, int ox, int oy,  int wiek, int BOARD_WIDTH, int BOARD_HEIGHT) {
        super(swiat, ox, oy, wiek, BOARD_WIDTH, BOARD_HEIGHT);
        this.nazwa = "Wilcze Jagody";
        this.inicjatywa = 0;
        this.sila = 0;
        this.znak = 'W';
        swiat.setJesliWolnePole(ox, oy, znak);
    }

    @Override
    public void zasiej(int x, int y) {
        swiat.addOrganizm(new wilcze_jagody(swiat,x, y, swiat.get_and_incrementKtoryUrodzony(),BOARD_WIDTH,BOARD_HEIGHT));
    }

    @Override
    public int kolizja(Organizm napotkany_org) {
        return POISON;
    }
}
